src/client/graphics/layers/RadialMenu.ts

```diff
@@ -267,7 +267,36 @@ export class RadialMenu implements Layer {
   }
 
   tick() {
-    // Update logic if needed
+    // Only update when menu is visible
+    if (this.isVisible && this.clickedCell) {
+      const myPlayer = this.g
+        .playerViews()
+        .find((p) => p.clientID() == this.clientID);
+      if (myPlayer && myPlayer.isAlive()) {
+        const tile = this.g.ref(this.clickedCell.x, this.clickedCell.y);
+        myPlayer.actions(tile).then((actions) => {
+          // Only update the boat option to avoid unnecessary processing
+          if (actions.canBoat) {
+            this.activateMenuElement(Slot.Boat, "#3f6ab1", boatIcon, () => {
```

Comment:
In the new context, `this.clickedCell` is a `null`able type again, so you should re-test it for null to avoid a potential runtime crash.

This is something that typescript's strictNullChecks mode would enforce, if it were enabled.

```suggestion
            this.activateMenuElement(Slot.Boat, "#3f6ab1", boatIcon, () => {
              if (this.clickedCell === null) return;
```